### PR TITLE
Feature/msbb

### LINKED ISSUE #49: Mosaic scene_bounding_box returns incorrect width and height.
On a mosaic image the scene bounding box width and height are set to the width and height of a single subblock rather than the entire scene.

It looks like the issue is in getSceneYXSize, it returns on the first subblock found for mosaic images.

I can work around this by iterating through every M_index and using read_subblock_rect to determine the min/max x/y returned but it is quite slow.

### PR DESCRIPTION
**Pull request by heeler**
This PR adds a requested feature to retrieve the bounding boxes for a scene within a mosaic file as a list of bounding boxes.

There is an added test to check this works and the C++ and python tests all pass.

resolves #49

There were formatting issues with PR50 that were causing havoc so I just added the code to a new branch.

### CODE REVIEW

**Comment by heeler** [comment]
@djdt and  @@sebi06 if you would let me know if this looks like it will do what you want.

File: _aicspylibczi/Reader.h [hunk 1]
```diff
@@ -263,32 +263,54 @@ namespace pylibczi {
...
+      }
 
       /*!
        * @brief get the shape of the loaded images
        * @param images_ the ImageVector of images to get the shape of
        * @param is_mosaic_ a boolean telling the function if it's a mosaic file
        * @return a vector of (Dimension letter, Dimension size)
        */
-      static Shape getShape(pylibczi::ImageVector& images_, bool is_mosaic_) { return images_.getShape(); }
+      static Shape getShape(pylibczi::ImageVector& images_, bool is_mosaic_)
```
Comment (toloudis) [inline _aicspylibczi/Reader.h:277]: is_mosaic is unused here

**Review by toloudis — APPROVED** [review]

**Comment by djdt** [comment]
This doesn't seem to limit the output bounding boxes to those with the specified scene. I get the same number of results with -1 and both valid and invalid scene indices for mosaic_scene_bounding_boxes.

**Comment by heeler** [comment]
@djdt Thanks, you're right I was just thinking about getting the bboxes out. I'll take a shot at fixing it later today.

**Comment by heeler** [comment]
Ok So I think this will do what you want now. I had made an assumption from a previous PR that conflicted with what I did here. Now it will return a list of the bounding boxes for a scene if the file has scenes and the index matches. 
It will return all the bounding boxes if Scene isn't defined in the file or the scene index is out of range.

**Comment by heeler** [comment]
@djdt and @sebi06 I think I have it working correctly now. By chance do either of you have a small multiscene mosaic file that you'd be willing to share I could add to the project for continuous integration tests? Ideally something around ~10MB or less.

**Comment by sebi06** [comment]
Sure. I upload one to Dropbox. One file is the CZI and the other one shows the scene layout.

https://www.dropbox.com/s/beitc1by0f7vop5/Multiscene_CZI_3Scenes.czi?dl=0

https://www.dropbox.com/s/r39dmx4myaqp8yx/Multiscene_CZI_3Scenes.png?dl=0

Let me know, if you cannot download it or you other CZIs

grafik

**Pull request merged**